src/Seq.Input.HealthCheck/HttpHealthCheck.cs

```diff
@@ -40,15 +43,32 @@ class HttpHealthCheck
         const int InitialContentChars = 16;
         const string OutcomeSucceeded = "succeeded", OutcomeFailed = "failed";
 
-        public HttpHealthCheck(HttpClient httpClient, string title, string targetUrl, JsonDataExtractor extractor, bool bypassHttpCaching)
+        public HttpHealthCheck(HttpClient httpClient, string title, string targetUrl, string authenticationHeaderValue, string optionalHeader, JsonDataExtractor extractor, bool bypassHttpCaching)
         {
             _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
             _title = title ?? throw new ArgumentNullException(nameof(title));
             _targetUrl = targetUrl ?? throw new ArgumentNullException(nameof(targetUrl));
+            _authenticationHeaderValue = authenticationHeaderValue;
+            _optionalHeader = optionalHeader;
             _extractor = extractor;
             _bypassHttpCaching = bypassHttpCaching;
         }
 
+        private static void AddOrOverwriteHeader(HttpRequestMessage request, string attribute, string value)
```

Comment:
HTTP calls them field "names", may be better to use `name` than `attribute` here